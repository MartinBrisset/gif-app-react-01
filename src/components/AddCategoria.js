import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategoria = ({setCategorias}) => {
    
    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        // console.log('enviado');
        
        if (inputValue.trim().length > 2) {
            setCategorias( categorias => [ inputValue, ...categorias ])
            setInputValue('')
        }
        
        
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input 
                    type='text'
                    value={inputValue}
                    onChange={ handleInputChange }
                    />
            </form>
        </>
    )
}

AddCategoria.propTypes = {
    setCategorias: PropTypes.func.isRequired
}
