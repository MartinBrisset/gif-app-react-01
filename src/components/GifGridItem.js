import React from 'react'

export const GifGridItem = ({id,title,url}) => {


    return (
        <div className='card animate__animated animate__fadeIn animate__delay-0.8s'>
            <img src={url} alt={title} />
            <p> {title} </p>
        </div>
    )
}
