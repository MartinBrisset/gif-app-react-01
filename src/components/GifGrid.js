import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({categoria}) => {

    const { loading, data:imgs } = useFetchGifs(categoria)

    return (
        <>
            <h3 className='animate__animated animate__fadeIn animate__delay-0.8s'> {categoria} </h3>

            {loading && <p className='animate__animated animate__flash'>Cargando...</p>}

            <div className='card-grid'>
                
                    { imgs.map( img => {
                        return <GifGridItem
                        key={img.id}
                        {...img} 
                        />
                    })}

            </div>
        
        </>

    )
}
