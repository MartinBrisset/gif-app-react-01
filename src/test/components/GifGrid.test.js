import React from "react"
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid  />', () => {

    const categoria = 'Goku'
    
    test('Debe de mostrar el componente correctamente ', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        
        const wrapper = shallow( <GifGrid categoria={categoria} />)
        expect( wrapper ).toMatchSnapshot()
        
    })
    
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa/cosa.jpg',
            title: 'Hola que tal'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow( <GifGrid categoria={categoria} />)

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists() ).toBe(false)
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )


    })
    

})