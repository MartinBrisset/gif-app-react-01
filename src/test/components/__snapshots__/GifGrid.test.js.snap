// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid  /> Debe de mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-0.8s"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid  /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-0.8s"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Hola que tal"
      url="https://localhost/cualquiercosa/cosa.jpg"
    />
  </div>
</Fragment>
`;
